feat(oss): add new streaming docs (#3763)
langchain-ai/docs: src/oss/deepagents/streaming-projections.mdx

---
title: Streaming
description: Stream subagents, messages, tool calls, and final output from Deep Agents.
---

> Deep Agents build on LangGraph's Projection Streaming model and add a first-class `subagents` projection. Use it when you want application-facing streams for the coordinator agent, delegated subagents, nested messages, tool calls, and final state.

Reviewer: I don't think describing it as a `subagents` projection is helpful to someone unless they are already familiar with the streaming model
```suggestion
Deep Agents build on LangGraph's Projection Streaming model and provide a way to stream only updates from `subagents`.
Use it when you want application-facing streams for the coordinator agent, delegated subagents, nested messages, tool calls, and final state.
```

<Note>

> Interested in lower-level Pregel stream modes and the existing Deep Agents streaming guide? See the [Mode Streaming page](/oss/deepagents/streaming).

Reviewer: ```suggestion
For lower-level Pregel stream modes or a guide on streaming for Deep Agents see the [Mode Streaming page](/oss/deepagents/streaming).
```

</Note>

## Stream subagents

> Deep Agents add a subagent projection on top of LangGraph streaming. Use `run.subagents` when you want one stream handle per delegated `task` call.

Reviewer: The abstract product
```suggestion
Deep Agents adds a subagent projection on top of LangGraph streaming. Use `run.subagents` when you want one stream handle per delegated `task` call.
```

:::python
```py
run = agent.stream_v2({
    "messages": [{"role": "user", "content": "Write me a haiku about the sea"}],
})

for subagent in run.subagents:
    print(subagent.name, subagent.path, subagent.status)

    for message in subagent.messages:
        print(str(message.text))
```
:::

:::js
```ts
const run = await agent.stream_v2({
  messages: [{ role: "user", content: "Write me a haiku about the sea" }],
});

for await (const subagent of run.subagents) {
  console.log(subagent.name);
  console.log(await subagent.taskInput);

  for await (const message of subagent.messages) {
    console.log(await message.text);
  }
}
```
:::

## Subagent stream fields

Each subagent stream is run-shaped, so you can recurse into it the same way you consume the parent run.

:::python
Python uses snake_case projection names such as `tool_calls`. Each subagent stream can expose `.messages`, `.tool_calls`, `.values`, `.subagents`, and `.output`.
:::

:::js
TypeScript uses camelCase projection names such as `toolCalls` and `taskInput`.
:::

| Field | Description |
| ----- | ----------- |
| `name` | Subagent name. |

> | `taskInput` | Prompt or input passed to the task tool. |

Reviewer: should this also have a python version?
